File: client/src/services/journalRecordController.js
import executeRequest from './executeRequest';

const journalRecordController = {
  create: student =>
    executeRequest('/journal', {
      method: 'POST',
      body: JSON.stringify(student),
      headers: {
        'Content-Type': 'application/json'
      }
    }),
  get: id => executeRequest(`/journal/${id}`),
  getAll: () => executeRequest('/journal'),
  getAllByStudyGroup: id => executeRequest(`/journal/study_group/${id}`),
  update: (id, student) =>
    executeRequest(`/journal/${id}`, {
      method: 'PUT',
      body: JSON.stringify(student),
      headers: {
        'Content-Type': 'application/json'
      }
    }),
  delete: id => executeRequest(`/journal/${id}`, { method: 'DELETE' })
};

export default journalRecordController;

File: server/src/main/java/com/lab/system/controller/JournalRecordController.java
package com.lab.system.controller;

import com.lab.system.dao.JournalRecordJdbc;
import com.lab.system.model.JournalRecord;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/journal")
public class JournalRecordController {

    private final JournalRecordJdbc journalRecordJdbc;

    public JournalRecordController(JournalRecordJdbc journalRecordJdbc) {
        this.journalRecordJdbc = journalRecordJdbc;
    }

    @GetMapping("")
    public List<JournalRecord> getJournalRecords() {
        return journalRecordJdbc.getAll();
    }

    @GetMapping("/{id}")
    public JournalRecord getJournalRecord(@PathVariable int id) {
        return journalRecordJdbc.get(id);
    }

    @GetMapping("/student/{studentId}")
    public List<JournalRecord> getJournalRecordsByStudent(@PathVariable int studentId) {
        return journalRecordJdbc.getAllByStudent(studentId);
    }

    @GetMapping("/study_group/{studyGroupId}")
    public List<JournalRecord> getJournalRecordsByStudyGroup(@PathVariable int studyGroupId) {
        return journalRecordJdbc.getAllByStudyGroup(studyGroupId);
    }

    @PostMapping(path = "", consumes = "application/json", produces = "application/json")
    public long addJournalRecord(@RequestBody JournalRecord journalRecord) {
        return journalRecordJdbc.create(journalRecord);
    }

    @PutMapping("/{id}")
    public int updateJournalRecord(@PathVariable int id, @RequestBody JournalRecord journalRecord) {
        journalRecordJdbc.update(id, journalRecord);
        return 0;
    }

    @DeleteMapping("/{id}")
    public int deleteJournalRecord(@PathVariable int id) {
        journalRecordJdbc.delete(id);
        return 0;
    }
}
